astropy/io/votable/tree.py

```diff
@@ -1675,7 +1675,8 @@ def parse(self, iterator, config):
     def to_xml(self, w, **kwargs):
         attrib = w.object_attrs(self, self._attr_list)
         if "unit" in attrib:
-            attrib["unit"] = self.unit.to_string("cds")
+            format = _get_unit_format(self._config)
+            attrib["unit"] = self.unit.to_string(format)
```

Comment:
Thank you for writing the PR!

An other issue pointed by Tom in the original thread was that the standard does not impose a format for the units, but only recommends it. So using any format (even out of `cds` and `vounit`) should raise (at most) a warning and not an error.

Maybe the call of `unit.to_string` should catch any `ValueError` and transform it into a warning?

Reply:
We could do that but I fear the side effects. Would be nice if we have a realistic use case of such violation that should be allowed to add to the test suite. Since I am not VO expert, do you have any suggestion on such a test? I would like to use it to make sure both read/write work. Thanks!